### PR TITLE
chore: ping network every 10 seconds to check connectivity

### PR DESCRIPTION
**Pull request by alexandreaco**
Screen Shot 2022-01-19 at 9 37 22 AM

### CODE REVIEW

**Review by staylorwr — CHANGES_REQUESTED** [review]
Thanks for this feature @alexandreaco!  It looks like this also highlights a
  regression in the refactor or error handling work where we're no longer able
  to visit the settings tabs to change the gateway / network without internet
  access.

  Can you work on that functionality in this PR (or document next steps towards bring that functionality back for the team)?

**Review by staylorwr — APPROVED** [review]

File: src/App.vue [hunk 1]
```diff
@@ -70,22 +72,28 @@ const App = defineComponent({
...
+        if (!connected.value) {
+          reloadSubscriptions()
+        }
+        setConnected(true)
+        setNetwork(network)
+      }).catch(() => {
+        if (connected.value) {
+          setConnected(false)
+          setNetwork(null)
+          setError(new Error(t('errors.networkError')))
```
Comment (saminakh) [inline src/App.vue:88]: 👍

File: src/views/Home/HomeEnterPasscode.vue [hunk 2]
```diff
@@ -74,23 +75,28 @@ const HomeEnterPasscode = defineComponent({
...
+        return firstValueFrom(sub)
       }).then((network) => {
         router.push('/wallet')
         setNetwork(network)
-        isAuthenticating.value = false
         walletLoaded()
       }).catch((error) => {
         setConnected(false)
-        if (error.cause === 'NETWORK_ID_FAILED') {
+        if (error.message === 'network timeout') {
```
Comment (saminakh) [inline src/views/Home/HomeEnterPasscode.vue:91]: Should we set an error here to show a modal letting the user know why they're getting sent straight to the node settings page? Perhaps a nice-to-have in the future?

Comment (saminakh) [reply src/views/Home/HomeEnterPasscode.vue:91]: On a similar note, the gateway change modal is a little wonky (`cancel` button not fitting). We should make an issue for this!
Screen Shot 2022-01-20 at 12 42 28 PM

**Review by saminakh — APPROVED** [review]
These changes LGTM and are working locally! Left a few thoughts for the future!

**Review by SK-Sam — APPROVED** [review]

**Pull request merged**